netket/models/equivariant.py

```diff
@@ -24,54 +24,168 @@
+from jax.scipy.special import logsumexp
 
 from netket import nn as nknn
-from netket.nn.initializers import zeros, variance_scaling
+from netket.nn.initializers import zeros
+from netket.nn.symmetric_linear import (
+    DenseSymmMatrix,
+    DenseSymmFFT,
+    DenseEquivariantFFT,
+    DenseEquivariantMatrix,
+    DenseEquivariantIrrep,
+)
 
 
-class GCNN(nn.Module):
-    r"""Implements a Group Convolutional Neural Network (G-CNN) that outputs a wavefunction
-    that is invariant over a specified symmetry group.
+def identity(x):
+    return x
 
-    The G-CNN is described in ` Cohen et. *al* <http://proceedings.mlr.press/v48/cohenc16.pdf>`_
-    and applied to quantum many-body problems in ` Roth et. *al* <https://arxiv.org/pdf/2104.05085.pdf>`_.
 
-    The G-CNN alternates convolution operations with pointwise non-linearities. The first
-    layer is symmetrized linear transform given by DenseSymm, while the other layers are
-    G-convolutions given by DenseEquivariant. The hidden layers of the G-CNN are related by
-    the following equation:
+def unit_normal_scaling(key, shape, dtype):
+    return jax.random.normal(key, shape, dtype) / jnp.sqrt(
+        jnp.prod(jnp.asarray(shape[1:]))
+    )
 
-    .. math ::
 
-        {\bf f}^{i+1}_h = \Gamma( \sum_h W_{g^{-1} h} {\bf f}^i_h).
+class GCNN_FFT(nn.Module):
+    """Implements a GCNN using a fast fourier transform over the translation group.
+    The group convolution can be written in terms of translational convolutions with
+    symmetry transformed filters as desribed in ` Cohen et. *al* <http://proceedings.mlr.press/v48/cohenc16.pdf>`_
+    The translational convolutions are then implemented with Fast Fourier Transforms.
+    """
 
+    symmetries: HashableArray
+    """A group of symmetry operations (or array of permutation indices) over which the network should be equivariant.
+    Numpy/Jax arrays must be wrapped into an :class:`netket.utils.HashableArray`.
+    """
+    product_table: HashableArray
+    """Product table describing the algebra of the symmetry group
+    Numpy/Jax arrays must be wrapped into an :class:`netket.utils.HashableArray`.
     """
+    shape: Tuple
+    """Shape of the translation group"""
+    layers: int
+    """Number of layers (not including sum layer over output)."""
+    features: Tuple
+    """Number of features in each layer starting from the input. If a single number is given,
+    all layers will have the same number of features."""
+    characters: HashableArray
+    """Array specifying the characters of the desired symmetry representation"""
+    dtype: Any = float
+    """The dtype of the weights."""
+    activation: Any = jax.nn.selu
+    """The nonlinear activation function between hidden layers."""
+    output_activation: Any = identity
+    """The nonlinear activation before the output. Defaults to the identity."""
+    imag_part: bool = False
+    """If true return only the imaginary part of the output"""
```

Comment:
You don't really return only the imaginary part, which would be a real number.
You set the real part to 0 and keep the imaginary part. because you return a complex number (If i understand correctly).
can you correct this docstring?

Reply:
I'll change the name to "equal_amplitude"

Reply:
and the docstring now states that we set Re[psi]=0 so that the amplitudes of all basis states equal

Reply:
it's `\Re[logpsi]` i think. I corrected it.